Mathlib/CategoryTheory/Quotient.lean

```diff
@@ -138,23 +136,23 @@ protected theorem sound {a b : C} {f₁ f₂ : a ⟶ b} (h : r f₁ f₂) :
   simpa using Quot.sound (CompClosure.intro (𝟙 a) f₁ f₂ (𝟙 b) h)
 #align category_theory.quotient.sound CategoryTheory.Quotient.sound
 
+@[simp]
+lemma compClosure_iff_self [h : Congruence r] {X Y : C} (f g : X ⟶ Y) :
+    CompClosure r f g ↔ r f g := by
+  constructor
+  · intro hfg
+    induction' hfg with m m' hm
+    exact Congruence.compLeft _ (Congruence.compRight _ (by assumption))
+  · exact CompClosure.of _ _ _
+
+theorem compClosure_eq_self [h : Congruence r] :
```

Comment:
Shouldn't this be the simp-lemma, instead of the one above?

Reply:
Yes, indeed, it should be better this way. I have just commited it.